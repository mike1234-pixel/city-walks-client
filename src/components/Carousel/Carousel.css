/* mdbreact corrections */

.carousel-container {
  min-width: 100%;
  padding: 0;
}

.z-depth-1 {
  box-shadow: none !important;
}

.carousel-inner {
  max-height: 400px;
}

.carousel-img {
  max-height: 400px;
}

/* end mdbreact corrections */
