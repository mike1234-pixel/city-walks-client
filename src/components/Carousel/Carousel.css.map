{"version":3,"sourceRoot":"","sources":["Carousel.scss"],"names":[],"mappings":"AAAA;AAAA;AAAA;AAAA;AAAA;AAMA;EACE;EACA;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;EACA;;;AAGF;AACA;EACE;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;AACE;EACA;EACA;;;AAGF;EACE;IACE;IACA;;;AAIJ","file":"Carousel.css"}