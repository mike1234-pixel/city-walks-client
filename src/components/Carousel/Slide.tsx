import {
  MDBCarouselCaption,
  MDBCarouselItem,
  MDBMask,
  MDBView,
} from "mdbreact";
import React from "react";

interface Props {
  slideNumber: number;
  imageSrc: string;
  imageDescription: string;
  title: string;
  description: string;
}

const Slide: React.FC<Props> = (props: Props) => {
  const { slideNumber, imageSrc, imageDescription, title, description } = props;

  return (
    <MDBCarouselItem itemId={slideNumber}>
      <MDBView className="carousel-view">
        <img
          className="d-block w-100 carousel-img"
          src={imageSrc}
          alt={imageDescription}
        />
        <MDBMask />
      </MDBView>
      <MDBCarouselCaption className="carousel-caption">
        <h3 className="h3-responsive">{title}</h3>
        <p>{description}.</p>
      </MDBCarouselCaption>
    </MDBCarouselItem>
  );
};

export default Slide;
