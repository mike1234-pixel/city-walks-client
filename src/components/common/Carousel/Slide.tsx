import { ReactNode } from "react";
import {
  MDBCarouselCaption,
  MDBCarouselItem,
  MDBMask,
  MDBView,
} from "mdbreact";

interface SlideProps {
  slideNumber: number;
  title: string;
  description: string;
  children?: ReactNode | ReactNode[];
}

const Slide = (props: SlideProps) => {
  const { slideNumber, title, description, children } = props;

  return (
    <MDBCarouselItem itemId={slideNumber}>
      <MDBView className="carousel-view">
        {children}
        <MDBMask />
      </MDBView>
      <MDBCarouselCaption className="carousel-caption">
        <h3 className="h3-responsive">{title}</h3>
        <p>{description}.</p>
      </MDBCarouselCaption>
    </MDBCarouselItem>
  );
};

export default Slide;
