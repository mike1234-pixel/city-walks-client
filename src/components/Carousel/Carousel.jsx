import React from "react"
import { MDBCarousel, MDBCarouselCaption, MDBCarouselInner, MDBCarouselItem, MDBView, MDBMask, MDBContainer } from "mdbreact"
import carouselImage1 from "./carousel-images/carousel-illustration-1-2.jpg"
import carouselImage2 from "./carousel-images/carousel-illustration-2-2.jpg"
import carouselImage3 from "./carousel-images/carousel-illustration-3-2.jpg"
import './Carousel.css'

const Carousel = () => {
  return (
    <MDBContainer className="carousel-container">
      <MDBCarousel
      activeItem={1}
      length={3}
      showControls={true}
      showIndicators={true}
      className="z-depth-1"
    >
      <MDBCarouselInner>
        <MDBCarouselItem itemId="1">
          <MDBView>
            <img
              className="d-block w-100 carousel-img"
              src={carouselImage1}
              alt="First slide"
            />
          <MDBMask />
          </MDBView>
          <MDBCarouselCaption>
            <h3 className="h3-responsive">Explore</h3>
            <p>First text</p>
          </MDBCarouselCaption>
        </MDBCarouselItem>
        <MDBCarouselItem itemId="2">
          <MDBView>
            <img
              className="d-block w-100 carousel-img"
              src={carouselImage2}
              alt="Second slide"
            />
          <MDBMask />
          </MDBView>
          <MDBCarouselCaption>
            <h3 className="h3-responsive">Strong mask</h3>
            <p>Second text</p>
          </MDBCarouselCaption>
        </MDBCarouselItem>
        <MDBCarouselItem itemId="3">
          <MDBView>
            <img
              className="d-block w-100 carousel-img"
              src={carouselImage3}
              alt="Third slide"
            />
          <MDBMask overlay="black-slight" />
          </MDBView>
          <MDBCarouselCaption>
            <h3 className="h3-responsive">Slight Mast</h3>
            <p>Third text</p>
          </MDBCarouselCaption>
        </MDBCarouselItem>
      </MDBCarouselInner>
    </MDBCarousel>
    </MDBContainer>
  );
}

export default Carousel;