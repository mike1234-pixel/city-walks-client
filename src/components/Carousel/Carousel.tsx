import React from "react"
import { MDBCarousel, MDBCarouselCaption, MDBCarouselInner, MDBCarouselItem, MDBView, MDBMask } from "mdbreact"
import carouselImage1 from "./assets/carousel-illustration-1-2.jpg"
import carouselImage2 from "./assets/carousel-illustration-2-2.jpg"
import carouselImage3 from "./assets/carousel-illustration-3-2.jpg"
import './Carousel.scss'

const Carousel: React.FC = () => {
  return (
    <div className="home-carousel">
      <MDBCarousel
        activeItem={1}
        length={3}
        showControls={true}
        showIndicators={true}
        className="z-depth-1"
      >
        <MDBCarouselInner>
          <MDBCarouselItem itemId="1">
            <MDBView className='carousel-view'>
              <img
                className="d-block w-100 carousel-img"
                src={carouselImage1}
                alt="First slide"
              />
              <MDBMask />
            </MDBView>
            <MDBCarouselCaption className="carousel-caption">
              <h3 className="h3-responsive">Explore</h3>
              <p>the history of cities, their landmarks and hidden gems.</p>
            </MDBCarouselCaption>
          </MDBCarouselItem>
          <MDBCarouselItem itemId="2">
            <MDBView className='carousel-view'>
              <img
                className="d-block w-100 carousel-img"
                src={carouselImage2}
                alt="Second slide"
              />
              <MDBMask />
            </MDBView>
            <MDBCarouselCaption className="carousel-caption">
              <h3 className="h3-responsive">Discover</h3>
              <p>local attractions, landmarks, museums and public spaces.</p>
            </MDBCarouselCaption>
          </MDBCarouselItem>
          <MDBCarouselItem itemId="3">
            <MDBView className='carousel-view'>
              <img
                className="d-block w-100 carousel-img"
                src={carouselImage3}
                alt="Third slide"
              />
              <MDBMask overlay="black-slight" />
            </MDBView>
            <MDBCarouselCaption className="carousel-caption">
              <h3 className="h3-responsive">Share</h3>
              <p>your experiences and recommendations.</p>
            </MDBCarouselCaption>
          </MDBCarouselItem>
        </MDBCarouselInner>
      </MDBCarousel>
    </div>
  );
}

export default Carousel;