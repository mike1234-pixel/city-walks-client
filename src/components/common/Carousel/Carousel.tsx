import { MDBCarousel, MDBCarouselInner } from "mdbreact";
import carouselImage1 from "./assets/carousel-illustration-1-2.jpg";
import carouselImage2 from "./assets/carousel-illustration-2-2.jpg";
import carouselImage3 from "./assets/carousel-illustration-3-2.jpg";
import Slide from "./Slide";
import "./Carousel.css";

const Carousel = () => {
  return (
    <div className='home-carousel'>
      <MDBCarousel
        activeItem={1}
        length={3}
        showControls={true}
        showIndicators={true}
        className='z-depth-1'
      >
        <MDBCarouselInner>
          <Slide
            slideNumber={1}
            title='Explore'
            description='the history of cities, their landmarks and hidden gems.'
          >
            <img
              className="d-block w-100 carousel-img"
              src={carouselImage1}
              alt='Shoreditch'
            />
          </Slide>
          <Slide
            slideNumber={2}
            title='Discover'
            description='local attractions, landmarks, museums and public spaces.'
          >
            <img
              className="d-block w-100 carousel-img"
              src={carouselImage2}
              alt='Edinburgh'
            />
          </Slide>
          <Slide
            slideNumber={3}
            title='Share'
            description='your experiences and recommendations.'
          >
            <img
              className="d-block w-100 carousel-img"
              src={carouselImage3}
              alt='Barbican'
            />
          </Slide>
        </MDBCarouselInner>
      </MDBCarousel>
    </div>
  );
};

export default Carousel;
