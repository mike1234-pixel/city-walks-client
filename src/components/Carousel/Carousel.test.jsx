import renderer from 'react-test-renderer'
import Carousel from './Carousel'

// snapshot test
it('Carousel renders correctly', () => {  

  const tree = renderer
    .create(<Carousel/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});