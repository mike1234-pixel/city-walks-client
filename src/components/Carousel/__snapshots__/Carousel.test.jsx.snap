// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel renders correctly 1`] = `
<div
  className="container carousel-container"
  data-test="container"
>
  <div
    aria-label="carousel"
    className="carousel carousel-fade z-depth-1"
    data-test="carousel"
    length={3}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="carousel-inner"
      data-test="carousel-inner"
    >
      <div
        className="carousel-item active"
        data-test="carousel-item"
        style={
          Object {
            "left": "0",
          }
        }
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <img
            alt="First slide"
            className="d-block w-100 carousel-img"
            src="https://images.pexels.com/photos/1121782/pexels-photo-1121782.jpeg?auto=compress&cs=tinysrgb&dpr=2&w=500"
          />
          <div
            className="mask rgba-black-light"
            data-test="mask"
          />
        </div>
        <div
          className="carousel-caption"
          data-test="carousel-caption"
        >
          <h3
            className="h3-responsive"
          >
            Explore
          </h3>
          <p>
            First text
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
        data-test="carousel-item"
        style={
          Object {
            "left": "0",
          }
        }
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <img
            alt="Second slide"
            className="d-block w-100 carousel-img"
            src="https://images.pexels.com/photos/460672/pexels-photo-460672.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
          />
          <div
            className="mask rgba-black-strong"
            data-test="mask"
          />
        </div>
        <div
          className="carousel-caption"
          data-test="carousel-caption"
        >
          <h3
            className="h3-responsive"
          >
            Strong mask
          </h3>
          <p>
            Second text
          </p>
        </div>
      </div>
      <div
        className="carousel-item"
        data-test="carousel-item"
        style={
          Object {
            "left": "0",
          }
        }
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <img
            alt="Third slide"
            className="d-block w-100 carousel-img"
            src="https://images.unsplash.com/photo-1471093304841-0423d25b23e5?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1050&q=80"
          />
          <div
            className="mask rgba-black-slight"
            data-test="mask"
          />
        </div>
        <div
          className="carousel-caption"
          data-test="carousel-caption"
        >
          <h3
            className="h3-responsive"
          >
            Slight Mast
          </h3>
          <p>
            Third text
          </p>
        </div>
      </div>
    </div>
    <a
      className="carousel-control-prev"
      data-slide="prev"
      data-test="carousel-control"
      onClick={[Function]}
    >
      <div>
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="sr-only"
        >
          Previous
        </span>
      </div>
    </a>
    <a
      className="carousel-control-next"
      data-slide="next"
      data-test="carousel-control"
      onClick={[Function]}
    >
      <div>
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="sr-only"
        >
          Next
        </span>
      </div>
    </a>
    <ol
      className="carousel-indicators"
      data-test="carousel-indicators"
    >
      <li
        className="active"
        data-test="carousel-indicator"
        onClick={[Function]}
      />
      <li
        className=""
        data-test="carousel-indicator"
        onClick={[Function]}
      />
      <li
        className=""
        data-test="carousel-indicator"
        onClick={[Function]}
      />
    </ol>
  </div>
</div>
`;
